service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/aws/s3storageFolder/ControlledAwsS3StorageFolderResource.java

```diff
@@ -115,6 +115,7 @@ public Optional<UniquenessCheckAttributes> getUniquenessCheckAttributes() {
     return Optional.of(
         new UniquenessCheckAttributes()
             .uniquenessScope(UniquenessScope.GLOBAL)
```

Comment:
- Yep, this will fail the check if any other S3 folder resource exists with the same bucket name and prefix in this WSM instance. This is only a check inside the WSM DB, it isn't calling out to AWS to validate bucket uniqueness (though we validate that later).
- Yes, this gets called as part of a single DB transaction that checks for uniqueness and also writes an initial DB entry. This is actually the first flight step for creating a controlled resource, but you're right that only one will move on with the flight and the other will fail immediately.

Reply:
for 2 - CreateControlledResourceFlight->StoreMetadataStep->createResourceStart(as a transaction) includes verifyUniqueness()